### DISCUSSION TITLE
Search criteria for room search and filters

### DISCUSSION THREAD

**Discussion by finnus** [Haus-des-Engagements/re-sharing#22] [Ideas]
Wrap up of our last discussion on search criteria:
- Users rather search for free slots than for specific rooms.
- the current teamup timetable is great for admins and regular users (like coworkers), but has some drawbacks for most users:
  - you can't see free slots but must deduce from booked slots when a room is still free
  - for that matter, the "planer" view is much better, but can become to big, if many rooms are available
- needed search options:
  - room capacity (max number of persons)
  - dates:
    - single date
    - date range (between this and this date) + weekdays selection (e.g. only Mondays or Thursdays)
  - time
    - fix starttime + duration 
    - time range for starttime + duration
- the search results can be shown in different ways but the most universal way (no matter what search criteria and how many results) would be a list with the following information per search result:
  - Room
  - date
  - start time + end time (e.g. 17:00-22:00, also if the duration is only set to 1h, as the user might then chose what fits best)

The booking of a room should then be possible from that results list. 

Nota bene:
- it would be nice to have a calender view as we have it right now for...
  - all rooms (by one organization), with de-selectable rooms
  - every single room (on the room page)
- it would be nice to be able to book the rooms directly from the calender view

**Comment by finnus** [comment]
@ToastMX: we can try the github discussion. feel free to add what I forgot!

**Comment by finnus** [comment]
This could be the search filter for the date selection:
![search_filter](https://github.com/Haus-des-Engagements/roomsharing/assets/40977905/58706ae9-1a12-4a4d-bfbc-334ab402e8f7)

**Comment by ToastMX** [comment]
Very good. 

Changing rooms within recurring bookings seems very useful.
There is a drwaback though, If really alot of bookings use this, this could lead to a ripple effect. Because every recurring booking blocks multiple regular recurring slots in multiple rooms. Which means other groups than also only find rooms with room jumping and so on. 
And i believe changing rooms is not ideal for the groups, its better than nothing, but not ideal.
